src/Raven.Server/Documents/Handlers/SubscriptionsHandler.cs

```diff
@@ -57,14 +58,33 @@ public class SubscriptionsHandler : DatabaseRequestHandler
                     writer.WriteStartObject();
                     writer.WritePropertyName("Results");
 
+                    
+                    
                     using (context.OpenReadTransaction())
                     {
-                        writer.WriteDocuments(context, 
-
-                            fetcher.GetDataToSend(context, state, etag, patch)
-                                .Where(x=>x.Data != null)
+                        // todo: return excepitons alongside witht the documents
+                        var dataToSend = fetcher.GetDataToSend(context, state, etag, patch);
+                        writer.WriteDocuments(context,
+                            dataToSend
+                                .Where(x=>x.exception == null).Select(x=>x.doc), false, out int _);
+                        
+                        writer.WriteComma();
+                        writer.WritePropertyName("Exceptions");
+                        writer.WriteStartArray();
+
+                        var first = true;
+                        
+                        foreach (var exceptionDetails in dataToSend.Where(x=>x.exception != null).Select(x=> 
```

Comment:
No, that is not a good idea. You are iterating over the data twice, and I'm not really sure why you want to do that.